include/exadg/postprocessor/pointwise_output_generator_base.h

```diff
@@ -0,0 +1,202 @@
+  using point_value_type = typename PointwiseOutputDataBase<dim>::point_value_type;
+
+  void
+  evaluate(VectorType const & solution, double const & time, int const & time_step_number);
+
+protected:
+  PointwiseOutputGeneratorBase(MPI_Comm const & comm);
+
+  virtual ~PointwiseOutputGeneratorBase() = default;
+
+  void
+  setup_base(dealii::DoFHandler<dim> const &      dof_handler_in,
+             dealii::Mapping<dim> const &         mapping_in,
+             PointwiseOutputDataBase<dim> const & pointwise_output_data_in);
+
+  void
+  add_scalar_quantity_dataset(std::string const & name);
+
+  void
+  add_vector_quantity_datasets(std::string const & name);
+
+  void
+  write_scalar_quantity(std::string const &         name,
+                        std::vector<Number> const & values,
+                        unsigned int const          first_component = 0)
+  {
+    AssertThrow(first_component == 0,
+                dealii::ExcMessage("This function can only be called with first_component==0."));
+
+    componentwise_result = std::move(dealii::Vector<Number>(values.begin(), values.end()));
+    write_component(name, componentwise_result);
+  }
+
+  template<int n_components>
+  void
+  write_scalar_quantity(std::string const &                                          name,
+                        std::vector<dealii::Tensor<1, n_components, Number>> const & values,
+                        unsigned int const first_component = 0)
+  {
+    this->extract_component(componentwise_result, values, first_component);
+    write_component(name, componentwise_result);
+  }
+
+  template<int n_components>
+  void
+  write_vectorial_quantity(std::string const &                                          name,
+                           std::vector<dealii::Tensor<1, n_components, Number>> const & values,
+                           unsigned int const first_component = 0)
+  {
+    for(unsigned int d = 0; d < dim; ++d)
+    {
+      this->extract_component(componentwise_result, values, d + first_component);
+      write_component(name + std::to_string(d), componentwise_result);
+    }
+  }
+
+  template<int n_components>
+  [[nodiscard]] std::vector<
+    typename dealii::FEPointEvaluation<n_components, dim, dim, Number>::value_type>
+  compute_point_values(VectorType const & solution) const
+  {
+    return dealii::VectorTools::point_values<n_components>(*remote_evaluator,
+                                                           *dof_handler,
+                                                           solution);
+  }
+
+private:
+  virtual void
+  do_evaluate(VectorType const & solution) = 0;
+
+  void
+  setup_remote_evaluator();
+
+  void
+  reinit_remote_evaluator();
+
+  void
+  create_hdf5_file();
+
+#ifdef DEAL_II_WITH_HDF5
+  void
+  add_evaluation_points_dataset(dealii::HDF5::Group & group, std::string const & name);
+#endif
+
+  void
+  fill_evaluation_points(std::string const & name);
+
+#ifdef DEAL_II_WITH_HDF5
```

Comment:
I would group together the two functions guarded by `DEAL_II_WITH_HDF5`.

Reply:
resolved with [93ec7fd](https://github.com/exadg/exadg/pull/255/commits/93ec7fd87353d984a30b73b260daaa2d6da69407)